src/nemo-application.c

```diff
@@ -1049,24 +905,7 @@ init_desktop (NemoApplication *self)
 	/* Initialize the desktop link monitor singleton */
 	nemo_desktop_link_monitor_get ();
 
-	if (!self->priv->no_desktop &&
```

Comment:
By the way, Nautilus has improved this a bit more and offers "force-desktop" along with "no-desktop"